test/LongFile.lean

```diff
@@ -20,49 +20,61 @@ The current value of 1500 does not exceed the allowed bound.
+-- Do not allow setting a `longFile` linter option if the file does not exceed the `defValue`
 set_option linter.style.longFile 1500
 
 /--
 warning: using 'exit' to interrupt Lean
 ---
-warning: The default value of the `longFile` linter is 1500.
-This file is 36 lines long which does not exceed the allowed bound.
-Please, remove the `set_option linter.style.longFile 1600`.
+warning: The default value of the `longFile` linter is 50.
+This file is 37 lines long which does not exceed the allowed bound.
+Please, remove the `set_option linter.style.longFile 60`.
 -/
 #guard_msgs in
 -- Do not allow unnecessarily increasing the `longFile` linter option
-set_option linter.style.longFile 1600 in
+set_option linter.style.longFileDefValue 50 in
+set_option linter.style.longFile 60 in
 #exit
 
 /--
 warning: using 'exit' to interrupt Lean
 ---
-warning: This file is 51 lines long, but the limit is 10.
+warning: This file is 51 lines long, but the limit is 20.
 
-You can extend the allowed length of the file using `set_option linter.style.longFile 1500`.
+You can extend the allowed length of the file using `set_option linter.style.longFile 200`.
 You can completely disable this linter by setting the length limit to `0`.
 -/
 #guard_msgs in
--- First, we silence the linter, so that we can set a default value smaller than 1500.
-set_option linter.style.longFile 0 in
--- Next, we test that the `longFile` linter warns when a file exceeds the allowed value.
-set_option linter.style.longFile 10 in
+-- We test that the `longFile` linter warns when a file exceeds the allowed value.
+set_option linter.style.longFileDefValue 10 in
+set_option linter.style.longFile 20 in
+#exit
+
+/-- warning: using 'exit' to interrupt Lean -/
+#guard_msgs in
+-- Check that the `candidate` value is allowed
+set_option linter.style.longFileDefValue 10 in
+set_option linter.style.longFile 200 in
+#exit
+
+/-- warning: using 'exit' to interrupt Lean -/
+#guard_msgs in
+-- Check that the `candidate - 100` value is allowed
+set_option linter.style.longFileDefValue 10 in
+set_option linter.style.longFile 100 in
 #exit
 
 /--
 warning: using 'exit' to interrupt Lean
 ---
-warning: The default value of the `longFile` linter is 1500.
-This file is 66 lines long which does not exceed the allowed bound.
-Please, remove the `set_option linter.style.longFile 1700`.
+warning: This file is 78 lines long. The current limit is 101, but it is expected to be 200:
+`set_option linter.style.longFile 200`.
+You can completely disable this linter by setting the length limit to `0`.
 -/
 #guard_msgs in
--- First, we silence the linter, so that we can set a default value smaller than 1500.
-set_option linter.style.longFile 0 in
--- If we set the allowed bound for the `longFile` linter that is too large,
--- the linter tells us to use a smaller bound.
-set_option linter.style.longFile 1700 in
+-- Check that a value different from `candidate` or `candidate - 100` value is not allowed
```

Comment:
As I said above: a test which makes the allowed option far too large would also be nice. Thanks!

Reply:
I added a test at the end, setting the `defValue` to 400 and the `linterBound` to 5000.